crates/server/src/services/data/map.rs

```diff
@@ -35,28 +35,25 @@ impl MapData {
         &self,
         root: &Hash<Sha256>,
         leaves: &[LogLeaf],
-    ) -> Result<MapProofBundle<Sha256, MapLeaf>, DataServiceError> {
+    ) -> Result<MapProofBundle<Sha256, LogId, MapLeaf>, DataServiceError> {
         let map = self
             .map_index
             .get(root)
             .ok_or_else(|| DataServiceError::RootNotFound(root.clone()))?;
-
         let mut proofs = Vec::new();
         for LogLeaf { log_id, record_id } in leaves {
-            let proof = map
-                .prove(log_id)
-                .ok_or_else(|| DataServiceError::PackageNotIncluded(log_id.clone()))?;
```

Comment:
I think we'd still want this early return of `DataServiceError::PackageNotIncluded` here as the log id is coming from the request and may not be part of the map, so I'd expect an error rather than the `unwrap` call below panicking.